### PR TITLE
[MODSOURMAN-1134] Fix No action status on entities match and no action

### PR DESCRIPTION
**Pull request by RomanChernetskyi**
## Purpose
When the entity matched and no action is performed, should be shown status 'No action' for the underlying entity
## Approach
- Modified JournalUtil logic to save journal_record for match action for marc bib/instance
- Modified get_job_log_entries and get_record_processing_log functions to return Discarded for MATCH journal record if no other record with other status exists

### CODE REVIEW

File: mod-source-record-manager-server/src/main/resources/templates/db_scripts/schema.json [hunk 1]
```diff
@@ -286,6 +286,16 @@
...
+    },
+    {
+      "run": "after",
+      "snippetPath": "create_get_job_log_entries_function.sql",
+      "fromModuleVersion": "mod-source-record-manager-3.9.0"
+    },
+    {
+      "run": "after",
+      "snippetPath": "create_get_record_processing_log_function.sql",
+      "fromModuleVersion": "mod-source-record-manager-3.9.0"
```
Comment (KaterynaSenchenko) [inline mod-source-record-manager-server/src/main/resources/templates/db_scripts/schema.json:298]: same here

File: mod-source-record-manager-server/src/main/resources/templates/db_scripts/schema.json [hunk 2]
```diff
@@ -286,6 +286,16 @@
       "run": "after",
       "snippetPath": "create_get_job_log_entries_function.sql",
       "fromModuleVersion": "mod-source-record-manager-3.8.0"
+    },
+    {
+      "run": "after",
+      "snippetPath": "create_get_job_log_entries_function.sql",
+      "fromModuleVersion": "mod-source-record-manager-3.9.0"
```
[the remark this answers is not in the document]
Comment (RomanChernetskyi) [reply mod-source-record-manager-server/src/main/resources/templates/db_scripts/schema.json]: Thanks, done

File: mod-source-record-manager-server/src/main/resources/templates/db_scripts/schema.json (hunk 1 above)
Comment (RomanChernetskyi) [reply mod-source-record-manager-server/src/main/resources/templates/db_scripts/schema.json]: Thanks, done

**Review by KaterynaSenchenko — APPROVED** [review]

**Review by Maksat-Galymzhan — APPROVED** [review]

File: mod-source-record-manager-server/src/main/resources/templates/db_scripts/schema.json [hunk 3]
```diff
@@ -286,6 +286,16 @@
       "run": "after",
       "snippetPath": "create_get_job_log_entries_function.sql",
       "fromModuleVersion": "mod-source-record-manager-3.8.0"
+    },
+    {
+      "run": "after",
+      "snippetPath": "create_get_job_log_entries_function.sql",
+      "fromModuleVersion": "mod-source-record-manager-3.8.2"
```
Comment (KaterynaSenchenko) [inline mod-source-record-manager-server/src/main/resources/templates/db_scripts/schema.json:293]:
```suggestion
      "fromModuleVersion": "mod-source-record-manager-3.8.3"
```

File: mod-source-record-manager-server/src/main/resources/templates/db_scripts/schema.json [hunk 4]
```diff
@@ -286,6 +286,16 @@
...
+    },
+    {
+      "run": "after",
+      "snippetPath": "create_get_record_processing_log_function.sql",
+      "fromModuleVersion": "mod-source-record-manager-3.8.2"
```
Comment (KaterynaSenchenko) [inline mod-source-record-manager-server/src/main/resources/templates/db_scripts/schema.json:298]:
```suggestion
      "fromModuleVersion": "mod-source-record-manager-3.8.3"
```

**Review by JavokhirAbdullayev — APPROVED** [review]
please update news md

**Pull request merged**